test/unit/function/sqr.cpp

```diff
@@ -0,0 +1,41 @@
+//======================================================================================================================
+/*
+  Kyosu - Complex Without Complexes
+  Copyright : KYOSU Contributors & Maintainers
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#include <kyosu/kyosu.hpp>
+#include <test.hpp>
+
+TTS_CASE_WITH ( "Check kyosu::sqr over real"
+              , kyosu::real_types
+              , tts::generate(tts::between(-10,10))
+              )
+(auto data)
+{
+  TTS_ULP_EQUAL(kyosu::sqr(data), eve::sqr(data), 0.5);
+};
+
+TTS_CASE_WITH ( "Check kyosu::sqr over complex"
+              , kyosu::real_types
+              , tts::generate(tts::between(-10,10), tts::between(-10,10))
+              )
+(auto r, auto i)
+{
+  auto c = kyosu::to_complex(r,i);
+  TTS_EXPECT(eve::all(kyosu::reldist(kyosu::sqr(c), c*c) <= 0.0001));
+};
+
+TTS_CASE_WITH ( "Check kyosu::sqr over quaternion"
+              , kyosu::real_types
+              , tts::generate ( tts::between(-10,10), tts::between(-10,10)
+                              , tts::between(-10,10), tts::between(-10,10)
+                              )
+              )
+<typename T>(T r, T i, T j, T k)
+{
+  using type = kyosu::as_quaternion_t<T>;
+  auto q = type(r,i,j,k);
+  TTS_EXPECT(eve::all(kyosu::reldist(kyosu::sqr(q), q*q) <= 0.0001));
```

Comment:
you should use the TTS_RELATIVE_EQUAL macro
If it doesn't work this means you didn't added the proper functions in tts in tests.hpp

Reply:
Fixed

Reply:
could not fix it